test/spec/modules/appnexusAstBidAdapter_spec.js

```diff
@@ -217,59 +188,91 @@ describe('AppNexusAdapter', () => {
-
-      const response = bidmanager.addBidResponse.firstCall.args[1];
-      expect(response).to.have.property('statusMessage', 'Bid available');
-      expect(response).to.have.property('cpm', 0.5);
+  })
+
+  describe('interpretResponse', () => {
+    let response = {
+      'version': '3.0.0',
+      'tags': [
+        {
+          'uuid': '3db3773286ee59',
+          'tag_id': 10433394,
+          'auction_id': '4534722592064951574',
+          'nobid': false,
+          'no_ad_url': 'http://lax1-ib.adnxs.com/no-ad',
+          'timeout_ms': 10000,
+          'ad_profile_id': 27079,
+          'ads': [
+            {
+              'content_source': 'rtb',
+              'ad_type': 'banner',
+              'buyer_member_id': 958,
+              'creative_id': 29681110,
+              'media_type_id': 1,
+              'media_subtype_id': 1,
+              'cpm': 0.5,
+              'cpm_publisher_currency': 0.5,
+              'publisher_currency_code': '$',
+              'client_initiated_ad_counting': true,
+              'rtb': {
+                'banner': {
+                  'content': '<!-- Creative -->',
+                  'width': 300,
+                  'height': 250
+                },
+                'trackers': [
+                  {
+                    'impression_urls': [
+                      'http://lax1-ib.adnxs.com/impression'
+                    ],
+                    'video_events': {}
+                  }
+                ]
+              }
+            }
+          ]
+        }
+      ]
+    };
+
+    it('should get correct bid response', () => {
+      let expectedResponse = [
+        {
+          'requestId': '3db3773286ee59',
+          'cpm': 0.5,
+          'creative_id': 29681110,
+          'dealId': undefined,
+          'width': 300,
+          'height': 250,
+          'ad': '<!-- Creative -->',
+          'mediaType': 'banner'
+        }
+      ];
+
+      let result = spec.interpretResponse(response);
+      expect(Object.keys(result[0])).to.deep.equal(Object.keys(expectedResponse[0]));
     });
 
     it('handles nobid responses', () => {
-      server.respondWith(JSON.stringify({
+      let response = {
         'version': '0.0.1',
         'tags': [{
           'uuid': '84ab500420319d',
           'tag_id': 5976557,
           'auction_id': '297492697822162468',
           'nobid': true
         }]
-      }));
-
-      adapter.callBids(REQUEST);
-      server.respond();
-      sinon.assert.calledOnce(bidmanager.addBidResponse);
+      };
 
-      const response = bidmanager.addBidResponse.firstCall.args[1];
-      expect(response).to.have.property(
-        'statusMessage',
-        'Bid returned empty or error response'
-      );
+      let result = spec.interpretResponse(response);
+      expect(result.length).to.equal(0);
     });
 
     it('handles non-banner media responses', () => {
-      server.respondWith(JSON.stringify({
+      // debugger; // eslint-disable-line
```

Comment:
Stray comment

Reply:
Damn, i tend to forget these debugger's a lot. Removing it.